tests/infer/test_predictive.py

```diff
@@ -0,0 +1,92 @@
+import torch
+
+import pyro
+import pyro.distributions as dist
+import pyro.optim as optim
+import pyro.poutine as poutine
+from pyro.infer.autoguide import AutoDelta, AutoDiagonalNormal
+from pyro.infer import SVI, TracePredictive, Trace_ELBO, predictive
+from tests.common import assert_close
+
+
+def model(num_trials):
+    with pyro.plate("data", num_trials.size(0)):
+        phi_prior = dist.Uniform(num_trials.new_tensor(0.), num_trials.new_tensor(1.))
+        success_prob = pyro.sample("phi", phi_prior)
+        return pyro.sample("obs", dist.Binomial(num_trials, success_prob))
+
+
+def one_hot_model(pseudocounts, classes=None):
+    probs_prior = dist.Dirichlet(pseudocounts)
+    probs = pyro.sample("probs", probs_prior)
+    with pyro.plate("classes", classes.size(0) if classes is not None else 1, dim=-1):
+        return pyro.sample("obs", dist.OneHotCategorical(probs), obs=classes)
+
+
+def beta_guide(num_trials):
+    phi_c0 = pyro.param("phi_c0", num_trials.new_tensor(5.0).expand([num_trials.size(0)]))
+    phi_c1 = pyro.param("phi_c1", num_trials.new_tensor(5.0).expand([num_trials.size(0)]))
```

Comment:
@ahmadsalim: I think this is probably similar to the 8 schools example we discussed? Once we have learnt `phi_c0` and `phi_c1`, it seems to me that changing the size of `num_trials` during prediction isn't a valid thing to do without adding further information in the guide - i.e. which local parameter to use for the new_trials? It is possible that we want to predict on only a subset of originally learned data-points. e.g. for the 8 schools case, say we want to do predictions only for schools - 1, 3, and 4. In that case. I think we will need to pass those indices to the guide explicitly, and not just randomly subsample from the learned data points. 

```python
def beta_guide(num_trials, indices=None):
    phi_c0 = pyro.param("phi_c0", num_trials.new_tensor(5.0).expand([num_trials.size(0)]))
    phi_c1 = pyro.param("phi_c1", num_trials.new_tensor(5.0).expand([num_trials.size(0)]))
    if indices is not None:
        phi_c0 = phi_c0[indices]
        phi_c1 = phi_c1[indices]
```

I think most likely you were using this as a simplifying example for a more complex use case that you have. Let me know if that's the case, and if I missed something.